examples/gno.land/p/moul/collection/collection.gno

```diff
@@ -0,0 +1,551 @@
+			// Rollback: remove from _id index
+			c.indexes[IDIndex].tree.Remove(idStr)
+			return 0
+		}
+
+		for _, key := range keys {
+			if idx.options&SparseIndex != 0 && key == "" {
+				continue
+			}
+			if idx.options&CaseInsensitiveIndex != 0 {
+				key = strings.ToLower(key)
+			}
+			storeIndex(idx, key, idStr)
+		}
+	}
+
+	return uint64(id)
+}
+
+// Get retrieves an object by index and key, returns an Entry or nil if not found.
+func (c *Collection) Get(indexName, key string) *Entry {
+	idx, exists := c.indexes[indexName]
+	if !exists {
+		return nil
+	}
+
+	if indexName == IDIndex {
+		obj, exists := idx.tree.Get(key)
+		if !exists {
+			return nil
+		}
+		return &Entry{ID: key, Obj: obj}
+	}
+
+	// For other indexes
+	if idx.options&CaseInsensitiveIndex != 0 {
+		key = strings.ToLower(key)
+	}
+
+	idData, exists := idx.tree.Get(key)
+	if !exists {
+		return nil
+	}
+
+	// If multi-value, just return the first object found
+	if idx.options&MultiValueIndex != 0 {
+		list, ok := idData.([]string)
+		if !ok || len(list) == 0 {
+			return nil
+		}
+		// Get the first ID from the slice
+		idStr := list[0]
+		obj, exists := c.indexes[IDIndex].tree.Get(idStr)
+		if !exists {
+			return nil
+		}
+		return &Entry{ID: idStr, Obj: obj}
+	}
+
+	// single-value index
+	idStr, ok := idData.(string)
+	if !ok {
+		return nil
+	}
+	obj, exists := c.indexes[IDIndex].tree.Get(idStr)
+	if !exists {
+		return nil
+	}
+	return &Entry{ID: idStr, Obj: obj}
+}
+
+// Entry represents a single object with its ID in the collection
+type Entry struct {
+	ID  string
+	Obj interface{}
+}
+
+// GetAll retrieves all entries matching the given key in the specified index.
+func (c *Collection) GetAll(indexName, key string) []Entry {
+	idx, exists := c.indexes[indexName]
+	if !exists {
+		return nil
+	}
+
+	if idx.options&CaseInsensitiveIndex != 0 {
+		key = strings.ToLower(key)
+	}
+
+	// Special handling for ID index
+	if indexName == IDIndex {
+		if obj, exists := idx.tree.Get(key); exists {
+			return []Entry{{ID: key, Obj: obj}}
+		}
+		return nil
+	}
+
+	idData, exists := idx.tree.Get(key)
+	if !exists {
+		return nil
+	}
+
+	// For multi-value indexes
+	if idx.options&MultiValueIndex != 0 {
+		if ids, ok := idData.([]string); ok {
+			result := make([]Entry, 0, len(ids))
+			for _, idStr := range ids {
+				if obj, exists := c.indexes[IDIndex].tree.Get(idStr); exists {
+					result = append(result, Entry{ID: idStr, Obj: obj})
+				}
+			}
+			return result
+		}
+		return nil
+	}
+
+	// For single-value indexes
+	if idStr, ok := idData.(string); ok {
+		if obj, exists := c.indexes[IDIndex].tree.Get(idStr); exists {
+			return []Entry{{ID: idStr, Obj: obj}}
+		}
+	}
+	return nil
+}
+
+// GetIndex returns the underlying tree for an index
+func (c *Collection) GetIndex(name string) avl.ITree {
+	idx, exists := c.indexes[name]
+	if !exists {
+		return nil
+	}
+	return idx.tree
+}
+
+// Delete removes an object by its ID
+func (c *Collection) Delete(id uint64) {
```

Comment:
Maybe change that to `Delete(id uint64) bool`? Might be useful to know if we deleted something or not.

Reply:
382b5d104caba7dccfb4754e69d4019b30f7898e